### ISSUE TITLE
GoLang: multiple-value this._io.ReadBitsInt() in single-value context

### ISSUE THREAD

**Issue by GreyCat** [kaitai-io/kaitai_struct#481]
Original by @malleblas was posted as https://github.com/kaitai-io/kaitai_struct_compiler/issues/146:

> I am using Kaitai Struct with the Go output generator. When doing lots of bit arrays, I get errors from the compiler reporting **multiple-value this._io.ReadBitsInt() in single-value context**.
> A snippet of the generated code is:
> ```
> 	this.Doors = make([]bool, 48)
> 	for i := range this.Doors {
> 		tmp26, err := this._io.ReadBitsInt(1) != 0
> 		if err != nil {
> 			return err
> 		}
> 		this.Doors[i] = tmp26
> ```
> 
> I changed it to the following to make it work:
> ```
> 	this.Doors = make([]bool, 48)
> 	for i := range this.Doors {
> 		tmp26, err := this._io.ReadBitsInt(1)
> 		if err != nil {
> 			return err
> 		}
> 		this.Doors[i] = tmp26 != 0
> ```
> 
> I am relatively new to GoLang and very new to Kaitai so I don't have much input in terms of how to fix the situation at the moment.
> 
> -Mike

**Comment by GreyCat** [comment]
@malleblas, can I ask you to post a .ksy file fragment which leads to generation of this code? I can guess that it's something like array of 48 bits, each read into separate element of the array?

**Comment by generalmimon** [comment]
As it turns out, this has been already fixed in the 0.9 compiler a while ago. This fragment

```yaml
meta:
  id: go_read_bits_int
seq:
  - id: doors
    type: b1
    repeat: expr
    repeat-expr: 48
```

gets compiled into this (using the [devel Web IDE](https://ide.kaitai.io/devel/)):

```go
	this.Doors = make([]bool, 48)
	for i := range this.Doors {
		tmp1, err := this._io.ReadBitsInt(1)
		if err != nil {
			return err
		}
		this.Doors[i] = tmp1 != 0
	}
```

The issue is resolved, so I'm closing it.

**Comment by generalmimon** [comment]
> As it turns out, this has been already fixed in the 0.9 compiler a while ago.

FWIW, more precisely in commit https://github.com/kaitai-io/kaitai_struct_compiler/commit/7da6d94dbd39c913ed6f456f081daec5576cb393 from April 21, 2018.
